syntax = "proto3";

package productpb;

option go_package = "github.com/ronaldoafonso/productinfo/productpb";


service ProductInfo {
    rpc addProduct(Product) returns (ProductID) {}
    rpc getProduct(ProductID) returns (Product) {}
    rpc addAllProducts(stream Product) returns (ProductIDs) {}
    rpc getAllProducts(Empty) returns (stream Product) {}
}

message ProductID {
    string value = 1;
}

message Product {
    string id = 1;
    string name = 2;
    string description = 3;
    float price = 4;
}

message ProductIDs {
    repeated ProductID productIDs = 1;
}

message Empty {}
